### PR TITLE
Positional Audio Update

### PR DESCRIPTION
**Pull request by capnlove**
- Audio Descriptions
- SceneObject Audio functions
- Scene Debug Mode "audio"
- Doppler Effect

### CODE REVIEW

File: engine/source/2d/sceneobject/SceneObject.cc [hunk 1]
```diff
@@ -256,6 +258,16 @@ SceneObject::~SceneObject()
         mpScene->removeFromScene( this );
     }
 
+    if (mAudioHandles.size())
+        {
```
Comment (dottools) [inline engine/source/2d/sceneobject/SceneObject.cc]: Strange codeblock indentation going on here. ;)

File: engine/source/2d/sceneobject/SceneObject.cc [hunk 2]
```diff
@@ -861,6 +887,24 @@ void SceneObject::sceneRenderOverlay( const SceneRenderState* sceneRenderState )
     {
         pScene->mDebugDraw.DrawSortPoint( getRenderPosition(), getSize(), mSortPoint );
     }
+
+    if (debugMask & Scene::SCENE_DEBUG_AUDIO_SOURCES)
+        {
```
Comment (dottools) [inline engine/source/2d/sceneobject/SceneObject.cc]: Again here too :/

File: engine/source/2d/sceneobject/SceneObject.cc [hunk 3]
```diff
@@ -4030,6 +4074,56 @@ bool SceneObject::writeField(StringTableEntry fieldname, const char* value)
...
+void SceneObject::refreshsources()
+{
+    if (mAudioHandles.size())
+    {
+        S32 index = 0;
+        for (typeAudioHandleVector::iterator itr = mAudioHandles.begin(); itr != mAudioHandles.end(); ++itr)
+        {
+            U32 handle = *itr;
+            
+            if (handle) {
```
Comment (dottools) [inline engine/source/2d/sceneobject/SceneObject.cc]: Suddenly coding style change and ending brace is indented far off to the right...
More strange codeblock indention going on lines 4118 - 4120.

File: engine/source/2d/sceneobject/SceneObject_ScriptBinding.h [hunk 4]
```diff
@@ -4469,4 +4472,116 @@ ConsoleMethodWithDocs(SceneObject, safeDelete, ConsoleVoid, 2, 2, ())
...
+
+/*! Stops a sound attached to a SceneObject
+@param Index - The Index of the sound (obtained via getSoundatIndex)
+*/
+ConsoleMethodWithDocs(SceneObject, stopSound, ConsoleVoid, 3, 3, (S32 index))
+{
+    const S32 index = dAtoi(argv[2]);
+    
+    if (!object->getSoundsCount())
+        {
```
Comment (dottools) [inline engine/source/2d/sceneobject/SceneObject_ScriptBinding.h]: Another one..

File: engine/source/audio/audio.cc [hunk 5]
```diff
@@ -872,6 +873,30 @@ AUDIOHANDLE alxCreateSource(const AudioAsset *profile, const MatrixF *transform)
 }
 
 //--------------------------------------------------------------------------
+AUDIOHANDLE alxCreateSource_AD(const AudioAsset *profile, const AudioDescription* description, const MatrixF *transform)
```
Comment (dottools) [inline engine/source/audio/audio.cc]: Indention 4 vs 8 spaces?

**Comment by capnlove** [comment]
Not going to be able to edit in the next few days. Thanks for the info!
At least, no tabs, right? :)

**Comment by capnlove** [comment]
How about now?

**Comment by dottools** [comment]
Sanity restored. ;)

Thanks Simon.

**Comment by greenfire27** [comment]
Wow!  Nice job Simon!

**Comment by yurembo** [comment]
Perfect!

**Pull request merged**